project-changed.py

```diff
@@ -10,12 +10,14 @@
     exit(1)
 
 with open('projects.yaml', 'r') as f:
-    projects = {x['project']: x for x in yaml.safe_load(f)}
+    dependency_map = {
+        x['project']: set(x.get('dependencies', []))
+        for x in yaml.safe_load(f)}
 
 orig_hash = sys.argv[1]
 target_project = sys.argv[2]
 
-if target_project not in projects.keys():
+if target_project not in dependency_map.keys():
```

Comment:
It seems it may be more efficient to use a dictionary lookup here, although for very small collections linear search may prove more efficient. Still, semantically didn’t lookup feels right?